data/templates/Detector_SpringCloudService.csx

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Threading;
+
+private static string GetQuery(OperationContext<ArmResource> cxt)
+{
+    return
+    $@"
+		let startTime = datetime({cxt.StartTime});
+		let endTime = datetime({cxt.EndTime});
+		cluster('ClusterName').database('DBName').YOUR_TABLE_NAME
+		| where Timestamp == startTime and Timestamp == endTime
+		YOUR_QUERY
+	";
+}
+
+
+[ArmResourceFilter(provider: "Microsoft.Microservices4Spring", resourceTypeName: "appClusters")]
+[Definition(Id = "YOUR_DETECTOR_ID", Name = "", Author = "YOUR_ALIAS", Description = "")]
+public async static Task<Response> Run(DataProviders dp, OperationContext<ArmResource> cxt, Response res)
+{
+    res.Dataset.Add(new DiagnosticData()
+    {
+        Table = await dp.Kusto.ExecuteClusterQuery(GetQuery(cxt)),
```

Comment:
Table = await dp.Kusto.ExecuteClusterQuery(GetQuery(cxt), null, "GetQuery") --> This will make sure they see the function name in DataSources tab.

Reply:
Ah right. Good Catch Nazeef!